old-tutorials-in-development/tutorials-in-development/R/dc-spatio-temporal-intro/06-CRS-formats-proj4-EPSG.Rmd

```diff
@@ -166,45 +224,114 @@ Note that there are no specified units above. This is because this geographic
+formats associated with that `CRS`.
 
 The codes and more information can be found on these websites:
-* <a href="http://www.epsg-registry.org/" target="_blank">The EPSG registry. </a>
-* <a href="http://spatialreference.org/" target="_blank">Spatialreference.org</a>
-* <a href="http://spatialreference.org/ref/epsg/" target="_blank">list of ESPG codes.</a>
+* <a href="http://www.epsg-registry.org/" target="_blank">The EPSG registry</a>
+* <a href="http://spatialreference.org/ref/epsg/" target="_blank">Spatial Reference's list of EPSG codes</a>
 
-``` {r crs-strings}
+You can also create a list of EPSG codes using the` make_epsg()` function in 
+the `rgdal` package in `R`. This can be useful if you need to look up a code 
+without using <a href="http://spatialreference.org/" target="_blank">Spatialreference.org</a>.
 
+``` {r EPSGcode-list}
+#load library
 library('rgdal')
+
+# create data frame of EPSG codes
 epsg = make_EPSG()
-# View(epsg)
+
+# View first rows of the data frame
 head(epsg)
 ```
 
-#### WKT or Well-known Text
-Well-known Text (WKT) allows for compact machine- and human-readable representation of
-geometric objects as well as to consisely describing the critical elements of
-coordinate reference system (CRS) definitions.
+Once you have a `data.frame` of EPSG definitions, you can search for a specific 
+code. Let’s search for 4326 to see what `CRS` it represents. We will use `dplyr` 
+pipes and the `filter()` function to quickly extract the row we want. Notice that 
+a double = sign (`==`) specifies equals to.
 
-The codes and more information can be found on these websites:
-* <a href="http://docs.opengeospatial.org/is/12-063r5/12-063r5.html#43" target="_blank">Open Geospatial Consortium WKT document. </a>
+``` {r code-search}
+# load library
+library(dplyr)
+
+# search and view proj4 string for the EPSG code 4326
+epsg %>%
+  filter(code == 4326)
+```
+
+Alternatively, you can search for geographic coordinate systems by searching 
+for `longlat` in the `prj4` column of the data frame. Use the `str_detect()` 
+from the `stringr` package in your pipe to find all `CRS` definitions that contain 
+the string `longlat`.
+
+``` {r longlat-search}
+# load library
+library(stringr)
+
+# search for longlat
+longlat <- epsg %>%
+  filter(str_detect(prj4, 'longlat'))
+
+# view first rows from search
+head(longlat)
+```
+
+Search for UTM projections the same way. 
+
+```{r utm-search}
+
+utm <- epsg %>%
+  filter(str_detect(prj4, 'utm'))
+
+head(utm)
+```
+
+
+## WKT or Well-known Text
+
+Well-known Text (WKT) allows for compact machine- and human-readable representation 
+of geometric objects. It concisely describes the critical elements of `CRS` 
+definitions using a combination of elements separated by commas (,) and brackets 
+([]). Elements are described in all caps. 
+
+Below is an example of WKT for the `CRS` information in EPSG code 4326:
+ 
+``` {r WKT}
```

Comment:
For each code chunk in the .Rmd you can define the type of code within the chunk. In this case, the WKT is not 'r' code, therefore this chunk will not run as written, and it will crash our 'knitting' code when converting this .Rmd file into the .md, .html, and .R files. Therefore, we can do a couple of things:

1. Remove this code chunk and just include the WKT with some 'code' formatting using HTML
2. Or better yet, just turn this code chunk 'off' using 'eval=F, purl=F'. We add the 'purl=F' to prevent this code chunk from appearing in the .R file because we don't ever want people to try to 'run' that text as r code (because it won't work).
Does that make sense?

Reply:
Yes! This does make sense. I wasn't sure what was the best way to display the WKT text. In [Leah's](https://www.earthdatascience.org/courses/earth-analytics/spatial-data-r/understand-epsg-wkt-and-other-crs-definition-file-types/) tutorial she put it into a code chunk so I copied what she did. I'll use option #2.